okhttp-urlconnection/src/main/java/okhttp3/internal/huc/OkHttpURLConnection.java

```diff
@@ -587,6 +588,32 @@ private void setProtocols(String protocolsString, boolean append) {
     }
   }
 
+  static final class UnexpectedException extends IOException {
+    static final Interceptor INTERCEPTOR = new Interceptor() {
+      @Override public Response intercept(Chain chain) throws IOException {
+        try {
+          return chain.proceed(chain.request());
+        } catch (IOException e) {
+          throw e;
+        } catch (Error | RuntimeException e) {
```

Comment:
Good catch, literally! I can't say I've caught an Error before, I assume that's ok.